source/common/quic/active_quic_listener.cc

```diff
@@ -294,6 +294,18 @@ ActiveQuicListenerFactory::ActiveQuicListenerFactory(
                                                        validation_visitor,
                                                        server_preferred_address_config_factory),
             validation_visitor, context_);
+
+    // This tells QUICHE to pretend that the client sent this connection option. In the current
+    // QUICHE implementation, the server preferred address will only be sent if the client sends
+    // this non-standard connection option, which is only sent by QUICHE clients. Setting this
+    // causes QUICHE to always send the server preferred address if configured to do so.
+    //
+    // TODO(ggreenway,RyanTheOptimist): Remove this when QUICHE has a better API for managing this
+    // behavior.
+    if (Runtime::runtimeFeatureEnabled(
+            "envoy.restart_features.quic_send_server_preferred_address_to_all_clients")) {
+      quic_config_.SetInitialReceivedConnectionOptions(quic::QuicTagVector{quic::kSPAD});
```

Comment:
FYI: Now that #32155 has landed, QUICHE has a well supported mechanism to enable server preferred address support without needing the SPAD copt. We can now call:

`SetQuicFlag(quic_always_support_server_preferred_address, true)`

I might suggest doing this in `refreshReloadableFlags()` from `runtime_impl.cc`.

Reply:
Oh nice! That will make it a reloadable instead of restartable flag, and be much less hacky. I'll update the PR to do it this way.